spring-pulsar/src/test/java/org/springframework/pulsar/core/SharedSubscriptionConsumerTests.java

```diff
@@ -0,0 +1,169 @@
+import org.springframework.pulsar.listener.PulsarContainerProperties;
+import org.springframework.pulsar.listener.PulsarRecordMessageListener;
+import org.springframework.pulsar.test.support.PulsarTestContainerSupport;
+
+/**
+ * Tests for shared subscription types in Pulsar consumer.
+ *
+ * @author Soby Chacko
+ */
+public class SharedSubscriptionConsumerTests implements PulsarTestContainerSupport {
+
+	@Test
+	void sharedSubscriptionRoundRobinBasicScenario() throws Exception {
+
+		Map<String, Object> config = Map.of("topicNames",
+				Collections.singleton("shared-subscription-single-msg-test-topic"), "subscriptionName",
+				"shared-subscription-single-msg-test-sub");
+
+		PulsarClient pulsarClient = PulsarClient.builder().serviceUrl(PulsarTestContainerSupport.getPulsarBrokerUrl())
+				.build();
+		DefaultPulsarConsumerFactory<String> pulsarConsumerFactory = new DefaultPulsarConsumerFactory<>(pulsarClient,
+				config);
+
+		CountDownLatch latch1 = new CountDownLatch(1);
+		CountDownLatch latch2 = new CountDownLatch(1);
+		CountDownLatch latch3 = new CountDownLatch(1);
+
+		PulsarContainerProperties pulsarContainerProperties1 = pulsarContainerProperties(latch1, "hello john doe",
+				SubscriptionType.Shared);
+		DefaultPulsarMessageListenerContainer<String> container1 = new DefaultPulsarMessageListenerContainer<>(
+				pulsarConsumerFactory, pulsarContainerProperties1);
+		container1.start();
+
+		PulsarContainerProperties pulsarContainerProperties2 = pulsarContainerProperties(latch2, "hello alice doe",
+				SubscriptionType.Shared);
+		DefaultPulsarMessageListenerContainer<String> container2 = new DefaultPulsarMessageListenerContainer<>(
+				pulsarConsumerFactory, pulsarContainerProperties2);
+		container2.start();
+
+		PulsarContainerProperties pulsarContainerProperties3 = pulsarContainerProperties(latch3, "hello buzz doe",
+				SubscriptionType.Shared);
+		DefaultPulsarMessageListenerContainer<String> container3 = new DefaultPulsarMessageListenerContainer<>(
+				pulsarConsumerFactory, pulsarContainerProperties3);
+		container3.start();
+
+		Map<String, Object> prodConfig = Map.of("topicName", "shared-subscription-single-msg-test-topic");
+		DefaultPulsarProducerFactory<String> pulsarProducerFactory = new DefaultPulsarProducerFactory<>(pulsarClient,
+				prodConfig);
+		PulsarTemplate<String> pulsarTemplate = new PulsarTemplate<>(pulsarProducerFactory);
+
+		pulsarTemplate.newMessage("hello john doe").sendAsync();
+		pulsarTemplate.newMessage("hello alice doe").sendAsync();
+		pulsarTemplate.newMessage("hello buzz doe").sendAsync();
+
+		Awaitility.await().atMost(Duration.ofSeconds(30)).until(() -> latch1.await(10, TimeUnit.SECONDS));
+		Awaitility.await().atMost(Duration.ofSeconds(30)).until(() -> latch2.await(10, TimeUnit.SECONDS));
+		Awaitility.await().atMost(Duration.ofSeconds(30)).until(() -> latch3.await(10, TimeUnit.SECONDS));
+
+		container1.stop();
```

Comment:
I know in the other listener tests we need to go back and put the resources in try-with-resources. I think it would be a good idea to do this now in this new test while there is only 3 tests in here. Wdyt?